core/base/v7/inc/ROOT/TDirectoryEntry.hxx

```diff
@@ -33,7 +31,7 @@ public:
 
 private:
   time_point_t fDate = clock_t::now(); ///< Time of last change
-  TClass* fType;
+  const std::type_info* fType;
```

Comment:
Even-though CastPointer no longer uses the TClass, it might makes sense to still keep it around.   Find it back from a type_info is not 'fast' (string lookup in a map-like container) and is necessary in some context (for example I/O).  On the other hand finding the type_info from a TClass is a constant operation ...